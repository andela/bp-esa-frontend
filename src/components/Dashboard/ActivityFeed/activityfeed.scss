.activity-feed-card {
  box-sizing: border-box;
  float: left;
  height: 385px;
  width: 50%;
  margin: 0 20px 20px 20px;
  border-radius: 5px;
  background-color: #FFF;
  border: 1px solid #EEF;
  overflow: hidden;

  .activity-feed-no-activity {
    position: relative;
    left: 35%;
    top: 35%;
    color: #667;
    font-size: 18px;
    letter-spacing: -0.1px;
    line-height: 22px;
  }

  .activity-feed-container{
    margin: 20px 0 20px 0;
    height: 46px;
    width: 500px;
    display: flex;
    flex-direction: row;
    animation: 1s fade-on-mount;

    @keyframes fade-on-mount {
      from {
        opacity: 0;
      }
      to {
        opacity: 1;
      }
    }

    .feeds-initials {
      height: 100%;
      width: 3em;
      border-radius: 50%;
      background-color: #DDE;
      color: #667;
      position: relative;
      display: flex;
      justify-content: center;
      align-items: center;
    }

    .feeds-content{
      padding-left: 1em;
      width: 100%;

      .feed-description {
        height: 22px;
        width: 31px;
        color: #667;
        font-size: 18px;
        font-weight: 300;
        letter-spacing: -0.1px;
        line-height: 22px;
        position: relative;
        top: 25%;

        .bold{
          font-weight: normal;
        }


        span {
          &.offboarding {
            color: #D32F2F;
          }

          &.onboarding {
            color: #43A047;
          }
        }
      }

      .feed-date{
        height: 15px;
        width: 113px;
        color: #889;
        font-size: 12px;
        letter-spacing: -0.07px;
        line-height: 15px;
        display: block;
      }

    }
  }
}
