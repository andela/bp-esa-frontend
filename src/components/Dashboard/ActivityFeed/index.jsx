import React from 'react';
import moment from 'moment';
import './activityfeed.scss';

/**
 * @description gets ordered initials of names
 * @param {string} fullNames from the API
 * @returns {string}
 */
const getInitials = (fullNames) => {
  const names = fullNames.split(',');
  const initials = `${names[1][1]}${names[0][0]}`;
  return (initials);
};

/**
 * @description orders names to appear as firstName and lastName
 * @param {string} fullNames takes unordered names from the API
 * @returns {string}
 */
const reorderNames = (fullNames) => {
  const namesArr = fullNames.split(',');
  const names = `${namesArr[1]} ${namesArr[0]}`;
  return (names);
};

const ActivityFeed = (props) => {
  const { reportData } = props;
  if (reportData.length === 0) {
    return (
      <div className="activity-feed-no-activity">No activities to display</div>
    );
  }
  return (
    reportData.map(data => (
      <div className="activity-feed-container" key={data.id}>
        <div className="feeds-initials">
          <span>{getInitials(data.fellowName)}</span>
        </div>
        <div className="feeds-content">
          <span className="feed-description">
            <span className="bold">{reorderNames(data.fellowName)}</span>
            <span> was </span>
            <span className={data.type === 'offboarding' ? 'offboarding' : 'onboarding'}>
              {data.type === 'offboarding' ? 'offboarded' : 'onboarded'}
            </span>
            <span>
              {data.type === 'offboarding' ? ' from ' : ' on '}
            </span>
            <span className="bold">{data.partnerName}</span>
          </span>
          <span className="feed-date">
            {moment(data.updatedAt).format('MM/DD/YYYY, HH:mm')}
          </span>
        </div>
      </div>
    ))
  );
};

export default ActivityFeed;
