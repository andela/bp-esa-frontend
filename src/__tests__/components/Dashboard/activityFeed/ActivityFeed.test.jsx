import React from 'react';
import { mount } from 'enzyme';
import ActivityFeed from '../../../../components/Dashboard/ActivityFeed/index';

describe('Activity Feed', () => {
  let wrapper;
  const props = {
    reportData: [
      {
        id: 1,
        fellowName: 'Kitika, Kelvin',
        type: 'offboarding',
        updatedAt: '2019-05-15T13:47:04.608Z',
      },
    ],
  };
  beforeEach(() => {
    wrapper = mount(<ActivityFeed {...props} />);
  });
  afterEach(() => {
    wrapper.unmount();
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('renders component with no content', () => {
    const newProps = {
      reportData: [],
    };
    const wrapperWithNewProps = mount(<ActivityFeed {...newProps} />);
    const noInfo = wrapperWithNewProps.find('.activity-feed-no-activity');
    expect(noInfo.length).toEqual(1);
    wrapperWithNewProps.unmount();
  });
  it('renders card fellow information', () => {
    const card = wrapper.find('.activity-feed-container');
    expect(card.length).toEqual(1);
  });
});
