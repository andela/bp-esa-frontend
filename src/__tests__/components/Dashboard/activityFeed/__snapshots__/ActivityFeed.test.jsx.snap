// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Activity Feed renders correctly 1`] = `
ReactWrapper {
  Symbol(enzyme.__unrendered__): <ActivityFeed
    reportData={
      Array [
        Object {
          "fellowName": "Kitika, Kelvin",
          "type": "offboarding",
          "updatedAt": "2019-05-15T13:47:04.608Z",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "getWrappingComponentRenderer": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "reportData": Array [
        Object {
          "fellowName": "Kitika, Kelvin",
          "type": "offboarding",
          "updatedAt": "2019-05-15T13:47:04.608Z",
        },
      ],
    },
    "ref": null,
    "rendered": Object {
      "instance": <div
        class="activity-feed-container"
      >
        <div
          class="feeds-initials"
        >
          <span>
            KK
          </span>
        </div>
        <div
          class="feeds-content"
        >
          <span
            class="feed-description"
          >
            <span
              class="bold"
            >
               Kelvin Kitika
            </span>
            <span>
               was 
            </span>
            <span
              class="offboarding"
            >
              offboarded
            </span>
            <span>
               from 
            </span>
            <span
              class="bold"
            />
          </span>
          <span
            class="feed-date"
          >
            05/15/2019, 14:47
          </span>
        </div>
      </div>,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="feeds-initials"
          >
            <span>
              KK
            </span>
          </div>,
          <div
            className="feeds-content"
          >
            <span
              className="feed-description"
            >
              <span
                className="bold"
              >
                 Kelvin Kitika
              </span>
              <span>
                 was 
              </span>
              <span
                className="offboarding"
              >
                offboarded
              </span>
              <span>
                 from 
              </span>
              <span
                className="bold"
              />
            </span>
            <span
              className="feed-date"
            >
              05/15/2019, 14:47
            </span>
          </div>,
        ],
        "className": "activity-feed-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": <div
            class="feeds-initials"
          >
            <span>
              KK
            </span>
          </div>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <span>
              KK
            </span>,
            "className": "feeds-initials",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": <span>
                KK
              </span>,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "KK",
              },
              "ref": null,
              "rendered": Array [
                "KK",
              ],
              "type": "span",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": <div
            class="feeds-content"
          >
            <span
              class="feed-description"
            >
              <span
                class="bold"
              >
                 Kelvin Kitika
              </span>
              <span>
                 was 
              </span>
              <span
                class="offboarding"
              >
                offboarded
              </span>
              <span>
                 from 
              </span>
              <span
                class="bold"
              />
            </span>
            <span
              class="feed-date"
            >
              05/15/2019, 14:47
            </span>
          </div>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="feed-description"
              >
                <span
                  className="bold"
                >
                   Kelvin Kitika
                </span>
                <span>
                   was 
                </span>
                <span
                  className="offboarding"
                >
                  offboarded
                </span>
                <span>
                   from 
                </span>
                <span
                  className="bold"
                />
              </span>,
              <span
                className="feed-date"
              >
                05/15/2019, 14:47
              </span>,
            ],
            "className": "feeds-content",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": <span
                class="feed-description"
              >
                <span
                  class="bold"
                >
                   Kelvin Kitika
                </span>
                <span>
                   was 
                </span>
                <span
                  class="offboarding"
                >
                  offboarded
                </span>
                <span>
                   from 
                </span>
                <span
                  class="bold"
                />
              </span>,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="bold"
                  >
                     Kelvin Kitika
                  </span>,
                  <span>
                     was 
                  </span>,
                  <span
                    className="offboarding"
                  >
                    offboarded
                  </span>,
                  <span>
                     from 
                  </span>,
                  <span
                    className="bold"
                  />,
                ],
                "className": "feed-description",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": <span
                    class="bold"
                  >
                     Kelvin Kitika
                  </span>,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": " Kelvin Kitika",
                    "className": "bold",
                  },
                  "ref": null,
                  "rendered": Array [
                    " Kelvin Kitika",
                  ],
                  "type": "span",
                },
                Object {
                  "instance": <span>
                     was 
                  </span>,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": " was ",
                  },
                  "ref": null,
                  "rendered": Array [
                    " was ",
                  ],
                  "type": "span",
                },
                Object {
                  "instance": <span
                    class="offboarding"
                  >
                    offboarded
                  </span>,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "offboarded",
                    "className": "offboarding",
                  },
                  "ref": null,
                  "rendered": Array [
                    "offboarded",
                  ],
                  "type": "span",
                },
                Object {
                  "instance": <span>
                     from 
                  </span>,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": " from ",
                  },
                  "ref": null,
                  "rendered": Array [
                    " from ",
                  ],
                  "type": "span",
                },
                Object {
                  "instance": <span
                    class="bold"
                  />,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                    "className": "bold",
                  },
                  "ref": null,
                  "rendered": Array [
                    undefined,
                  ],
                  "type": "span",
                },
              ],
              "type": "span",
            },
            Object {
              "instance": <span
                class="feed-date"
              >
                05/15/2019, 14:47
              </span>,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "05/15/2019, 14:47",
                "className": "feed-date",
              },
              "ref": null,
              "rendered": Array [
                "05/15/2019, 14:47",
              ],
              "type": "span",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "reportData": Array [
          Object {
            "fellowName": "Kitika, Kelvin",
            "type": "offboarding",
            "updatedAt": "2019-05-15T13:47:04.608Z",
          },
        ],
      },
      "ref": null,
      "rendered": Object {
        "instance": <div
          class="activity-feed-container"
        >
          <div
            class="feeds-initials"
          >
            <span>
              KK
            </span>
          </div>
          <div
            class="feeds-content"
          >
            <span
              class="feed-description"
            >
              <span
                class="bold"
              >
                 Kelvin Kitika
              </span>
              <span>
                 was 
              </span>
              <span
                class="offboarding"
              >
                offboarded
              </span>
              <span>
                 from 
              </span>
              <span
                class="bold"
              />
            </span>
            <span
              class="feed-date"
            >
              05/15/2019, 14:47
            </span>
          </div>
        </div>,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="feeds-initials"
            >
              <span>
                KK
              </span>
            </div>,
            <div
              className="feeds-content"
            >
              <span
                className="feed-description"
              >
                <span
                  className="bold"
                >
                   Kelvin Kitika
                </span>
                <span>
                   was 
                </span>
                <span
                  className="offboarding"
                >
                  offboarded
                </span>
                <span>
                   from 
                </span>
                <span
                  className="bold"
                />
              </span>
              <span
                className="feed-date"
              >
                05/15/2019, 14:47
              </span>
            </div>,
          ],
          "className": "activity-feed-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": <div
              class="feeds-initials"
            >
              <span>
                KK
              </span>
            </div>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <span>
                KK
              </span>,
              "className": "feeds-initials",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": <span>
                  KK
                </span>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "KK",
                },
                "ref": null,
                "rendered": Array [
                  "KK",
                ],
                "type": "span",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": <div
              class="feeds-content"
            >
              <span
                class="feed-description"
              >
                <span
                  class="bold"
                >
                   Kelvin Kitika
                </span>
                <span>
                   was 
                </span>
                <span
                  class="offboarding"
                >
                  offboarded
                </span>
                <span>
                   from 
                </span>
                <span
                  class="bold"
                />
              </span>
              <span
                class="feed-date"
              >
                05/15/2019, 14:47
              </span>
            </div>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="feed-description"
                >
                  <span
                    className="bold"
                  >
                     Kelvin Kitika
                  </span>
                  <span>
                     was 
                  </span>
                  <span
                    className="offboarding"
                  >
                    offboarded
                  </span>
                  <span>
                     from 
                  </span>
                  <span
                    className="bold"
                  />
                </span>,
                <span
                  className="feed-date"
                >
                  05/15/2019, 14:47
                </span>,
              ],
              "className": "feeds-content",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": <span
                  class="feed-description"
                >
                  <span
                    class="bold"
                  >
                     Kelvin Kitika
                  </span>
                  <span>
                     was 
                  </span>
                  <span
                    class="offboarding"
                  >
                    offboarded
                  </span>
                  <span>
                     from 
                  </span>
                  <span
                    class="bold"
                  />
                </span>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <span
                      className="bold"
                    >
                       Kelvin Kitika
                    </span>,
                    <span>
                       was 
                    </span>,
                    <span
                      className="offboarding"
                    >
                      offboarded
                    </span>,
                    <span>
                       from 
                    </span>,
                    <span
                      className="bold"
                    />,
                  ],
                  "className": "feed-description",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": <span
                      class="bold"
                    >
                       Kelvin Kitika
                    </span>,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": " Kelvin Kitika",
                      "className": "bold",
                    },
                    "ref": null,
                    "rendered": Array [
                      " Kelvin Kitika",
                    ],
                    "type": "span",
                  },
                  Object {
                    "instance": <span>
                       was 
                    </span>,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": " was ",
                    },
                    "ref": null,
                    "rendered": Array [
                      " was ",
                    ],
                    "type": "span",
                  },
                  Object {
                    "instance": <span
                      class="offboarding"
                    >
                      offboarded
                    </span>,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "offboarded",
                      "className": "offboarding",
                    },
                    "ref": null,
                    "rendered": Array [
                      "offboarded",
                    ],
                    "type": "span",
                  },
                  Object {
                    "instance": <span>
                       from 
                    </span>,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": " from ",
                    },
                    "ref": null,
                    "rendered": Array [
                      " from ",
                    ],
                    "type": "span",
                  },
                  Object {
                    "instance": <span
                      class="bold"
                    />,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": undefined,
                      "className": "bold",
                    },
                    "ref": null,
                    "rendered": Array [
                      undefined,
                    ],
                    "type": "span",
                  },
                ],
                "type": "span",
              },
              Object {
                "instance": <span
                  class="feed-date"
                >
                  05/15/2019, 14:47
                </span>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "05/15/2019, 14:47",
                  "className": "feed-date",
                },
                "ref": null,
                "rendered": Array [
                  "05/15/2019, 14:47",
                ],
                "type": "span",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
